### DISCUSSION TITLE
How to test before merging to main?

### DISCUSSION THREAD

**Discussion by jonapich** [renovatebot/renovate#16940] [Request Help] [answered]
I am adding new definitions to my `renovate.json` file in a branch. I want to test this new configuration, simulate some dry runs to confirm the behavior, then open a PR for my colleagues to review and approve before I actually merge to main.

The problem is that the package I'm trying to configure is already up-to-date in main. I want to simulate (dry run) what would happen if the package is outdated. So naturally I tried to downgrade the version in my branch, but renovate doesn't seem to care about that at all, it will clone main and analyze that, period.

I tried to play with the "baseBranches" configuration element but it doesn't seem to serve this purpose at all.

How can I ensure my new configuration works without downgrading the package in the default branch?

For context, I am trying to run it locally like this, from my branch:

```
RENOVATE_TOKEN=... RENOVATE_CONFIG_FILE=renovate.json LOG_LEVEL=debug npx renovate org/repo --dry-run
```

The file I want to update looks like this in my default branch:

```
ARG RUNNER_VERSION=2.294.0
```

It was changed to the following in my branch to add the metadata info and to downgrade for simulation:

```
# renovate: datasource=github-releases depName=actions/runner
ARG RUNNER_VERSION=2.293.0
```

But since renovate doesn't look at my branch, it can't find the `renovate: ` comment and thus, finds no package.

**Comment by Gabriel-Ladzaretti** [comment, marked as answer]
Renovate will only scan branches specified in the `baseBranches` array, therefore your branch should be added to it.
you can then check the `PackageFiles with updates` in the logs to see if your dependency is being detected. 

https://docs.renovatebot.com/configuration-options/#basebranches
Note that `baseBranches` doesn't work with `forkMode` if set.

**Reply by jonapich** [reply]
Thanks, it worked. I guess I had something else going on in my previous attempts that prevented the configuration to be read properly.
